tests/functional/UseCase/Table/Import/FromTable/ImportTableFromTableTest.php

```diff
@@ -691,4 +695,128 @@ public function testLoadNullsToRequiredFields(): void
             ));
         }
     }
+
+    public function importTypeProvide(): Generator
+    {
+        yield 'full' => [ImportOptions\ImportType::FULL];
+        yield 'incremental' => [ImportOptions\ImportType::INCREMENTAL];
+    }
+
+    /**
+     * @dataProvider importTypeProvide
+     */
+    public function testLoadDataToDifferentColumnTypeEndsWithMismatchException(int $importType): void
+    {
+        $sourceTableName = $this->getTestHash() . '_Test_table';
+        $destinationTableName = $this->getTestHash() . '_Test_table_final';
+        $bucketDatabaseName = $this->bucketResponse->getCreateBucketObjectName();
+        $bqClient = $this->clientManager->getBigQueryClient($this->testRunId, $this->projectCredentials);
+
+        // create tables
+        $tableSourceDef = new BigqueryTableDefinition(
+            $bucketDatabaseName,
+            $sourceTableName,
+            false,
+            new ColumnCollection([
+                BigqueryColumn::createGenericColumn('id'),
+                BigqueryColumn::createGenericColumn('price'),
+            ]),
+            [],
+        );
+        $qb = new BigqueryTableQueryBuilder();
+        $sql = $qb->getCreateTableCommand(
+            $tableSourceDef->getSchemaName(),
+            $tableSourceDef->getTableName(),
+            $tableSourceDef->getColumnsDefinitions(),
+            $tableSourceDef->getPrimaryKeysNames(),
+        );
+        $bqClient->runQuery($bqClient->query($sql));
+        $insert = [];
+        // data in source table have nulls -> should fail, because we are trying to fit null value in to required field
```

Comment:
Žádné nully v tom insertu nejsou, ne?

Reply:
Jinak ten test mi přijde OK (byť možná vkládat tam víc řádků je zbytečné - mohls tam napsat celou query jako string a vložit jeden řádek).